### ISSUE TITLE
Deepen stack of pending changes

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#8770]
### Describe the problem

Right now Weblate tracks pending state per unit - whether current state needs to be written out to the file. This causes performance issues in several places, as pending changes might have to be written out before doing other edits:

* Accepting and editing suggestion triggers commit
* Bulk edit triggers commit
* Two users editing translation at same time triggers a lot of commits
* Add-ons working on a single translation can cause a lot of commits

### Describe the solution you'd like

* Use `Change` for tracking pending changes, add any missing info to `details` there, `Unit` would need just `pending_changes = models.ManyToManyField("Change")`.
* Utilize this to commit pending changes
* Drop written out relations

### Describe alternatives you've considered

* Decicated model tracking pending changes (unit, target, state, author, timestamp) could be used instead of relying on `Change`

### Additional context

* Should be designed with https://github.com/WeblateOrg/weblate/issues/3745 in mind
* User `user_commit_pending` should be refactored to use new model
* Not sure what to do with the pending flag, it would still simplify the lookup, but it duplicates the information, leading to possible consistency issues. Dropping field on the table might be expensive during the migration.

**Comment by nijel** [comment]
@amCap1712 This is a bigger one and will affect several locations. Let's approach it in steps so that we can agree on the design decisions before implementing them. The solution described in the issue doesn't have to be the best one. The biggest constraint to consider is that adding/removing field to models with many objects (such as `Unit` or `Change`) tends to be expensive on large instances and should be considered carefully.

**Comment by nijel** [comment]
The intention is to get rid of commits while editing. So commit from this code path should be removed.

**Comment by amCap1712** [comment]
Yes, I understand that. But when the commit actually happens should it
commit all the changes to a unit since last commit or just the latest
change?

**Comment by amCap1712** [comment]
I realized we can't do that because if a user changed the explanation for a unit and another the target text, the changes could be mis-attributed in the commits if we combined the changes into a single commit.

**Comment by nijel** [comment]
The commits should be grouped by author. Probably existing logic can be reused, it just spans across units now.
